pointer_test.go

```diff
@@ -78,13 +78,13 @@ func Test_mainCoalescer_coalescePointer(t *testing.T) {
 			"cycle 1",
 			simpleCycle(),
 			(*cycle)(nil),
-			simpleCycle(), // unnoticed
+			&cycle{Cycle: &cycle{}},
```

Comment:
Cycle detection changed behavior a bit because of deep-copy. But it still works as before, barring the issues raised in #24.